type_description_interfaces/msg/FieldType.msg

```diff
@@ -0,0 +1,157 @@
+uint8 FIELD_TYPE_WCHAR_ARRAY = 62
+uint8 FIELD_TYPE_BOOLEAN_ARRAY = 63
+uint8 FIELD_TYPE_BYTE_ARRAY = 64
+uint8 FIELD_TYPE_STRING_ARRAY = 65
+uint8 FIELD_TYPE_WSTRING_ARRAY = 66
+uint8 FIELD_TYPE_FIXED_STRING_ARRAY = 67
+uint8 FIELD_TYPE_FIXED_WSTRING_ARRAY = 68
+uint8 FIELD_TYPE_BOUNDED_STRING_ARRAY = 69
+uint8 FIELD_TYPE_BOUNDED_WSTRING_ARRAY = 70
+
+# Bounded Sequence Types
+uint8 FIELD_TYPE_NESTED_TYPE_BOUNDED_SEQUENCE = 97
+uint8 FIELD_TYPE_INT8_BOUNDED_SEQUENCE = 98
+uint8 FIELD_TYPE_UINT8_BOUNDED_SEQUENCE = 99
+uint8 FIELD_TYPE_INT16_BOUNDED_SEQUENCE = 100
+uint8 FIELD_TYPE_UINT16_BOUNDED_SEQUENCE = 101
+uint8 FIELD_TYPE_INT32_BOUNDED_SEQUENCE = 102
+uint8 FIELD_TYPE_UINT32_BOUNDED_SEQUENCE = 103
+uint8 FIELD_TYPE_INT64_BOUNDED_SEQUENCE = 104
+uint8 FIELD_TYPE_UINT64_BOUNDED_SEQUENCE = 105
+uint8 FIELD_TYPE_FLOAT_BOUNDED_SEQUENCE = 106
+uint8 FIELD_TYPE_DOUBLE_BOUNDED_SEQUENCE = 107
+uint8 FIELD_TYPE_LONG_DOUBLE_BOUNDED_SEQUENCE = 108
+uint8 FIELD_TYPE_CHAR_BOUNDED_SEQUENCE = 109
+uint8 FIELD_TYPE_WCHAR_BOUNDED_SEQUENCE = 110
+uint8 FIELD_TYPE_BOOLEAN_BOUNDED_SEQUENCE = 111
+uint8 FIELD_TYPE_BYTE_BOUNDED_SEQUENCE = 112
+uint8 FIELD_TYPE_STRING_BOUNDED_SEQUENCE = 113
+uint8 FIELD_TYPE_WSTRING_BOUNDED_SEQUENCE = 114
+uint8 FIELD_TYPE_FIXED_STRING_BOUNDED_SEQUENCE = 115
+uint8 FIELD_TYPE_FIXED_WSTRING_BOUNDED_SEQUENCE = 116
+uint8 FIELD_TYPE_BOUNDED_STRING_BOUNDED_SEQUENCE = 117
+uint8 FIELD_TYPE_BOUNDED_WSTRING_BOUNDED_SEQUENCE = 118
+
+# Unbounded Sequence Types
+uint8 FIELD_TYPE_NESTED_TYPE_UNBOUNDED_SEQUENCE = 145
+uint8 FIELD_TYPE_INT8_UNBOUNDED_SEQUENCE = 146
+uint8 FIELD_TYPE_UINT8_UNBOUNDED_SEQUENCE = 147
+uint8 FIELD_TYPE_INT16_UNBOUNDED_SEQUENCE = 148
+uint8 FIELD_TYPE_UINT16_UNBOUNDED_SEQUENCE = 149
+uint8 FIELD_TYPE_INT32_UNBOUNDED_SEQUENCE = 150
+uint8 FIELD_TYPE_UINT32_UNBOUNDED_SEQUENCE = 151
+uint8 FIELD_TYPE_INT64_UNBOUNDED_SEQUENCE = 152
+uint8 FIELD_TYPE_UINT64_UNBOUNDED_SEQUENCE = 153
+uint8 FIELD_TYPE_FLOAT_UNBOUNDED_SEQUENCE = 154
+uint8 FIELD_TYPE_DOUBLE_UNBOUNDED_SEQUENCE = 155
+uint8 FIELD_TYPE_LONG_DOUBLE_UNBOUNDED_SEQUENCE = 156
+uint8 FIELD_TYPE_CHAR_UNBOUNDED_SEQUENCE = 157
+uint8 FIELD_TYPE_WCHAR_UNBOUNDED_SEQUENCE = 158
+uint8 FIELD_TYPE_BOOLEAN_UNBOUNDED_SEQUENCE = 159
+uint8 FIELD_TYPE_BYTE_UNBOUNDED_SEQUENCE = 160
+uint8 FIELD_TYPE_STRING_UNBOUNDED_SEQUENCE = 161
+uint8 FIELD_TYPE_WSTRING_UNBOUNDED_SEQUENCE = 162
+uint8 FIELD_TYPE_FIXED_STRING_UNBOUNDED_SEQUENCE = 163
+uint8 FIELD_TYPE_FIXED_WSTRING_UNBOUNDED_SEQUENCE = 164
+uint8 FIELD_TYPE_BOUNDED_STRING_UNBOUNDED_SEQUENCE = 165
+uint8 FIELD_TYPE_BOUNDED_WSTRING_UNBOUNDED_SEQUENCE = 166
+
+# Identifying number for the type of the field, using one of the above constants.
+uint8 type_id
+
+# Only used when the type is an array or a bounded sequence.
+# In the case of an array, this is the fixed length of the array.
+# In the case of a bounded sequence, this is the maximum size of the sequence.
+# In all other cases this field is unused.
+uint64 length
```

Comment:
I think `length` might be confusing here. `maximum_size` would be a good alternative.

```suggestion
uint64 capacity
```

Reply:
I'm not sure if max\_size or capacity would give off the impression that a fixed length array could be shortened though, but I don't mind using `capacity` too

Reply:
Changed to capacity in 2563dc382581359b57d945b3a8b321599e215030